src/coco/repository/util/repositoryUtil.ts

```diff
@@ -0,0 +1,86 @@
+import RepositoryType from "../model/RepositoryType";
+import Repository from "../model/Repository";
+import { execSync } from "child_process";
+import Logger from "../../../log/Logger";
+
+const GET_GIT_REPOSITORY_COMMAND = "git config --local --get remote.origin.url";
+const GIT_REPO_URL_SUFFIX = ".git";
+const DOMAIN_GITHUB = "github.com";
+const DOMAIN_GITLAB = "gitlab.com";
+const DOMAIN_BITBUCKET = "bitbucket.org";
+const CUSTOM_DOMAIN_REG_EXP_GITHUB = /github.(.+).com/; // TODO
+const SUFFIX_HTTPS = "https://";
+
+function getRepositoryForType(rootFolderPath: string, type: RepositoryType): Repository {
+    let fullUrl: string | undefined;
+    try {
+        fullUrl = execSync(GET_GIT_REPOSITORY_COMMAND, {
```

Comment:
doesn't this freeze the UI?

Reply:
No, its just a file read operation which is what we do all he time in the project